### PR TITLE
Fix javadoc, sources and distribution generation.

### PR DESCRIPTION
**Pull request by baldimir**
Some problems showed up in fullProfile builds (nightlies) after migration to Apache parent pom. 
- Fixes javadoc generation.
- Fixes sources jars generation. 
- Fixes Drools distribution zip generation.

### CODE REVIEW

**Comment by baldimir** [comment]
Moving to Draft, I just found out there is duplicate Javadoc Plugin declaration in the build pom. Will fix that.

**Comment by baldimir** [comment]
Moved to Ready for review.

**Review by jstastny-cz — APPROVED** [review]
LGTM.

File: drools-alphanetwork-compiler/pom.xml [hunk 1]
```diff
@@ -36,7 +36,7 @@
   <name>Drools :: Alpha Network Compiler</name>
 
   <properties>
-    <java.module.name>org.drools.core.alphanetworkcompiler</java.module.name>
+    <java.module.name>org.drools.core.alphanetwork.compiler</java.module.name>
```
Comment (lucamolteni) [inline drools-alphanetwork-compiler/pom.xml]: Package is `org.drools.ancompiler` without `core` in the module, perhaps we should use that?

**Review by lucamolteni — APPROVED** [review]

File: drools-alphanetwork-compiler/pom.xml (hunk 1 above)
Comment (baldimir) [reply drools-alphanetwork-compiler/pom.xml]: Yes, makes sense, I overlooked this. Will make a fix and push. Thank you for noticing this.

Comment (baldimir) [reply drools-alphanetwork-compiler/pom.xml]: Fixed.

**Comment by kie-ci3** [comment]
**PR job** `#3` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-drools/${BRANCH:main}/.ci/buildchain-config.yaml' -o 'bc' -p apache/incubator-kie-drools -u https://github.com/apache/incubator-kie-drools/pull/5541 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/drools/job/main/job/pullrequest_jobs/job/drools-pr/job/PR-5541/3/display/redirect

**Test results:**
- PASSED: 27892
- FAILED: 4

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/drools/job/main/job/pullrequest_jobs/job/drools-pr/job/PR-5541/3/testReport/org.kie.kogito.index.oracle/ProcessDataIndexOracleHttpIT/testProcessInstanceEvents/">org.kie.kogito.index.oracle.ProcessDataIndexOracleHttpIT.testProcessInstanceEvents</a></summary>
java.util.concurrent.CompletionException: java.lang.RuntimeException: Unable to start Quarkus test resource class org.kie.kogito.index.test.quarkus.http.DataIndexOracleHttpQuarkusTestResource
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/drools/job/main/job/pullrequest_jobs/job/drools-pr/job/PR-5541/3/testReport/org.kie.kogito.index.oracle/ProcessDataIndexOracleKafkaIT/testProcessInstanceEvents/">org.kie.kogito.index.oracle.ProcessDataIndexOracleKafkaIT.testProcessInstanceEvents</a></summary>
java.util.concurrent.CompletionException: java.lang.RuntimeException: Unable to start Quarkus test resource class org.kie.kogito.index.test.quarkus.kafka.DataIndexOracleQuarkusKafkaTestResource
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/drools/job/main/job/pullrequest_jobs/job/drools-pr/job/PR-5541/3/testReport/org.kie.kogito.index/ProcessDataIndexMongoDBIT/testProcessInstanceEvents/">org.kie.kogito.index.ProcessDataIndexMongoDBIT.testProcessInstanceEvents</a></summary>
Failed to load ApplicationContext
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/drools/job/main/job/pullrequest_jobs/job/drools-pr/job/PR-5541/3/testReport/org.kie.kogito.index/ProcessDataIndexOracleIT/testProcessInstanceEvents/">org.kie.kogito.index.ProcessDataIndexOracleIT.testProcessInstanceEvents</a></summary>
Failed to load ApplicationContext
</details>

**Comment by baldimir** [comment]
The PR fail is not related to these changes. Merging.

**Pull request merged**